05.wc/wc.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def parse_options
+  options = { lines: false, words: false, bytes: false }
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: wc.rb [options] [file ...]'
+    opts.on('-l', 'Count lines') { options[:lines] = true }
+    opts.on('-w', 'Count words') { options[:words] = true }
+    opts.on('-c', 'Count bytes') { options[:bytes] = true }
+  end.parse!
+  options.transform_values! { |_| true } unless options.values.any?
+  options
+end
+
+def count_lines(content)
+  content.count("\n")
+end
+
+def count_words(content)
+  content.split(/\s+/).reject(&:empty?).count
+end
+
+def count_bytes(content)
+  content.bytesize
+end
+
+def output_results(counts, options, filename = nil, total: false)
+  output = []
+  output << format('%8d', counts[:lines]) if options[:lines]
+  output << format('%8d', counts[:words]) if options[:words]
+  output << format('%8d', counts[:bytes]) if options[:bytes]
+  output << " #{filename}" if filename
+  output << ' total' if total && filename.nil?
```

Comment:
```suggestion
  output << counts[:filename]
```

countsのハッシュの中にfilenameも格納しておくとコードがスッキリします。
合計の場合は "total" が入るイメージです